show/src/lib.rs

```diff
@@ -73,8 +91,71 @@ impl Args for Options {
                 show_proj_id,
                 show_ssh_key,
                 show_profile_id,
+                urn,
             },
             vec![],
         ))
     }
 }
+
+#[derive(Serialize, Debug)]
+enum Payload {
+    Person(payload::PersonPayload),
+    Project(payload::ProjectPayload),
+}
+
+pub fn run(mut options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let sock = keys::ssh_auth_sock();
+    let (_, storage) = keys::storage(&profile, sock)?;
+
+    if options == Options::default() {
+        options.show_proj_id = true;
+        options.show_peer_id = true;
+        options.show_self = true;
+        options.show_profile_id = true;
+        options.show_ssh_key = true;
+    }
+
+    if let Some(urn) = options.urn {
+        let payload = project::get_identity(&storage, &urn)
+            .map(|o| o.map(|p| p.payload().to_owned()).map(Payload::Project))
+            .or_else(|_| {
+                person::get_identity(&storage, &urn)
+                    .map(|o| o.map(|p| p.payload().to_owned()).map(Payload::Person))
+            })
+            .map_err(|_: anyhow::Error| anyhow::anyhow!("Couldn't load project or person."))?
+            .ok_or(anyhow::anyhow!("No project or person found for urn"))?;
```

Comment:
Have you tried to use `librad::git::identities::any::get`? It returns an enum of `Person/Project`, so you don't have to do this manually.

Reply:
you'd get an output like this:
```
{
  "https://radicle.xyz/link/identities/person/v1": {
    "name": "username"
  }
}
```

whereas if you map to the owned enum, you'd get:
```
{
  "Person": {
    "https://radicle.xyz/link/identities/person/v1": {
      "name": "username"
    }
  }
}
```
https://github.com/radicle-dev/radicle-client-tools/commit/d336edc2d66e49f544cd91b6459f92805f982548